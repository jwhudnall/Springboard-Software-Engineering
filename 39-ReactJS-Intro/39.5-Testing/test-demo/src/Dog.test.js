import { render, screen } from "@testing-library/react";
import Dog from "./Dog";

test("renders without crashing", () => {
  render(<Dog name='Millie' isAdopted={true} />);
});

it("Should match snapshot", () => {
  const { asFragment } = render(<Dog name='Millie' isAdopted={true} />);
  expect(asFragment()).toMatchSnapshot();
});

it("Should match snapshot (false)", () => {
  const { asFragment } = render(<Dog name='Millie' isAdopted={false} />);
  expect(asFragment()).toMatchSnapshot();
});
