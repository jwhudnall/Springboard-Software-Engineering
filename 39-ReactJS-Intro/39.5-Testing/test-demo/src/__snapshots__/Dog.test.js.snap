// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot (false) 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Millie
    </h1>
    <button>
      Adopt me!
    </button>
  </div>
</DocumentFragment>
`;

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Millie
    </h1>
    <p>
      Millie has been adopted!
    </p>
  </div>
</DocumentFragment>
`;
